agent/api/container/container.go

```diff
@@ -990,6 +990,25 @@ func (c *Container) GetLogDriver() string {
 	return hostConfig.LogConfig.Type
 }
 
+// GetNetworkModeFromHostConfig returns the network mode used by the container from the host config .
+func (c *Container) GetNetworkModeFromHostConfig() string {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	if c.DockerConfig.HostConfig == nil {
+		return ""
+	}
+
+	hostConfig := &dockercontainer.HostConfig{}
+	err := json.Unmarshal([]byte(*c.DockerConfig.HostConfig), hostConfig)
+	if err != nil {
+		seelog.Warnf("Encountered error when trying to get network mode for container %s: %v", err)
+		return ""
```

Comment:
this assumes that it is default network mode, which is ok for now. can you add a TODO to return err, string from this method to distinguish between error and default mode?

Reply:
I added the todo.